applications/tari_base_node/src/builder.rs

```diff
@@ -365,12 +365,37 @@ where
     )
     .await;
     debug!(target: LOG_TARGET, "Base node service registration complete.");
+
     let outbound_interface = handles
         .get_handle::<OutboundNodeCommsInterface>()
-        .expect("Problem getting node interface handle");
+        .expect("Problem getting node interface handle.");
     let chain_metadata_service = handles
         .get_handle::<ChainMetadataHandle>()
-        .expect("Problem getting chain metadata interface handle");
+        .expect("Problem getting chain metadata interface handle.");
+
+    // TODO: The local base node public key should be provided to the Transaction service and Output manager service,
+    // but this will require the local base node to be in the peer manager. This is a temporary fix where a random peer
+    // base node is provided.
+    if let Some(base_node_peer) = comms
+        .peer_manager()
+        .random_peers(1)
+        .expect("No peers in peer manager.")
+        .first()
```

Comment:
```suggestion
    if let Some(base_node_peer) = comms
        .peer_manager()
        .random_peers(1)
        .unwrap()
        .first()
```

`random_peers` fails if there is some lmdb error - if there are no peers it just returns an empty `Vec`